### PR TITLE
MM-52175 - fix button formatting and add anchor to the end

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
fix button formatting for the onboarding task completed screen `"got it"` button  and add anchor to the end to fix the "download our apps" link.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-52291
https://mattermost.atlassian.net/browse/MM-52175

#### Screenshots

https://user-images.githubusercontent.com/10082627/233389160-37f386e7-473a-4816-8e2d-2e534751c1ec.mov

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/onboarding_tasklist/onboarding_tasklist_completed.tsx [hunk 1]
```diff
@@ -239,7 +242,8 @@ const Completed = (props: Props): JSX.Element => {
                                     link: (msg: React.ReactNode) => (
                                         <ExternalLink
                                             location='onboarding_tasklist_completed'
-                                            href='https://mattermost.com/download/#desktop'
+                                            href='https://mattermost.com/download'
+                                            anchor='#desktop'
```
Comment (julmondragon) [inline webapp/channels/src/components/onboarding_tasklist/onboarding_tasklist_completed.tsx:246]: nit: Maybe the `#` should be appended in the `ExternalLink` component when `anchor` prop is provided.

**Review by julmondragon — APPROVED** [review]
Thanks @pvev

**Review by JulienTant — APPROVED** [review]
LGTM Pablo!

**Comment by pvev** [comment]
@stevemudie friendly reminder on this one 👍

File: webapp/channels/src/components/external_link/index.tsx [hunk 2]
```diff
@@ -52,7 +53,11 @@ export default function ExternalLink(props: Props) {
             // If the href already has query params, remove them before adding them back with the addition of the new ones
             href = href?.split('?')[0];
         }
-        href = `${href}?${queryString}`;
+        let anchor = '';
+        if (props.anchor) {
+            anchor = props.anchor.startsWith('#') ? props.anchor : `#${props.anchor}`;
+        }
+        href = `${href}?${queryString}${anchor}`;
```
[the remark this answers is not in the document]
Comment (pvev) [reply webapp/channels/src/components/external_link/index.tsx:60]: could you please provide more details on how your suggestion should work?

[the remark this answers is not in the document]
Comment (pvev) [reply webapp/channels/src/components/external_link/index.tsx:60]: oh, never mind, I think I see what you mean @nickmisasi , let me give it a try.

**Comment by pvev** [comment]
@nickmisasi updated with your suggested changes
Screenshot 2023-04-26 at 16 01 07

**Comment by pvev** [comment]
@nickmisasi @stevemudie friendly reminder on this one :)

**Review by nickmisasi — APPROVED** [review]
Code LGTM. 

Is that amount of whitespace in the card expected? Seems like a lot?
image

**Comment by annelieseklein** [comment]
@pvev i think we originally added the "got it" button to dismiss the task list forever, but can we just dismiss it if they click the X at this point or anywhere outside the task list? If so, I'd like to make the primary button here the download apps cta actually, like this: https://www.figma.com/file/N9M2op728km1Ve5LE8v3xe/Untitled?node-id=39-22795&t=PSZbOTnSBrr1QBiA-11

Screen Shot 2023-04-27 at 10 49 41 AM

**Comment by pvev** [comment]
> @pvev i think we originally added the "got it" button to dismiss the task list forever, but can we just dismiss it if they click the X at this point or anywhere outside the task list? If so, I'd like to make the primary button here the download apps cta actually, like this: https://www.figma.com/file/N9M2op728km1Ve5LE8v3xe/Untitled?node-id=39-22795&t=PSZbOTnSBrr1QBiA-11
> 
> Screen Shot 2023-04-27 at 10 49 41 AM

thanks @annelieseklein ok, this can be handled as a different ticket and we can take it for an upcoming iteration.
cc: @marianunez

**Comment by stevemudie** [comment]
Good news and bad news @pvev 

The good news is the link works as expected and the button is formatted correctly.

The bad news is along the way I found that the link to the apps within the task list just goes to the download page so it needs the anchor.

Can you fix that here or would you like me to write up a new ticket?

Also, sorry for having missed that before now. 🤦🏻‍♂️

**Review by stevemudie — APPROVED** [review]
Looks good @pvev Thank you for the additional fix to the task list link!

**Pull request merged**